tests/components/filter/test_sensor.py

```diff
@@ -15,309 +16,307 @@
-        assert 18.05 == filtered.state
-
-    def test_range(self):
-        """Test if range filter works."""
-        lower = 10
-        upper = 20
-        filt = RangeFilter(
-            entity=None, precision=2, lower_bound=lower, upper_bound=upper
-        )
-        for unf_state in self.values:
-            unf = float(unf_state.state)
-            filtered = filt.filter_state(unf_state)
-            if unf < lower:
-                assert lower == filtered.state
-            elif unf > upper:
-                assert upper == filtered.state
-            else:
-                assert unf == filtered.state
-
-    def test_range_zero(self):
-        """Test if range filter works with zeroes as bounds."""
-        lower = 0
-        upper = 0
-        filt = RangeFilter(
-            entity=None, precision=2, lower_bound=lower, upper_bound=upper
-        )
-        for unf_state in self.values:
-            unf = float(unf_state.state)
-            filtered = filt.filter_state(unf_state)
-            if unf < lower:
-                assert lower == filtered.state
-            elif unf > upper:
-                assert upper == filtered.state
-            else:
-                assert unf == filtered.state
-
-    def test_throttle(self):
-        """Test if lowpass filter works."""
-        filt = ThrottleFilter(window_size=3, precision=2, entity=None)
-        filtered = []
-        for state in self.values:
-            new_state = filt.filter_state(state)
-            if not filt.skip_processing:
-                filtered.append(new_state)
-        assert [20, 21] == [f.state for f in filtered]
-
-    def test_time_throttle(self):
-        """Test if lowpass filter works."""
-        filt = TimeThrottleFilter(
-            window_size=timedelta(minutes=2), precision=2, entity=None
-        )
-        filtered = []
-        for state in self.values:
-            new_state = filt.filter_state(state)
-            if not filt.skip_processing:
-                filtered.append(new_state)
-        assert [20, 18, 22] == [f.state for f in filtered]
-
-    def test_time_sma(self):
-        """Test if time_sma filter works."""
-        filt = TimeSMAFilter(
-            window_size=timedelta(minutes=2), precision=2, entity=None, type="last"
-        )
-        for state in self.values:
+
+def test_precision_zero(values):
+    """Test if precision of zero returns an integer."""
+    filt = LowPassFilter(window_size=10, precision=0, entity=None, time_constant=10)
+    for state in values:
+        filtered = filt.filter_state(state)
+    assert isinstance(filtered.state, int)
+
+
+def test_lowpass(values):
+    """Test if lowpass filter works."""
+    filt = LowPassFilter(window_size=10, precision=2, entity=None, time_constant=10)
+    out = ha.State("sensor.test_monitored", "unknown")
+    for state in [out] + values + [out]:
+        try:
             filtered = filt.filter_state(state)
-        assert 21.5 == filtered.state
```

Comment:
and this ?

Reply:
Same thing, this `filtered = filt.filter_state(state)` is in `test_time_sma` which is a few tests below this, as is the `assert 21.5 == filtered_state`.